File: ant-sdk/antd-php/src/Models/FilePutResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of a private file upload. The DataMap is returned to the caller;
 * it is NOT stored on-network.
 */
readonly class FilePutResult
{
    public function __construct(
        /** Hex-encoded caller-held DataMap. */
        public string $dataMap,
        /** Total storage cost paid in token units (atto). "0" if all chunks already existed. */
        public string $storageCostAtto,
        /** Total gas cost paid in wei as a decimal string. */
        public string $gasCostWei,
        /** Number of chunks stored on the network (uint64). */
        public int $chunksStored,
        /** Which payment mode was actually used: "auto", "merkle", or "single". */
        public string $paymentModeUsed,
    ) {}
}

File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/FilePutResult.java
package com.autonomi.antd.models;

/**
 * Result of a private file upload. The DataMap is returned to the caller;
 * it is NOT stored on-network.
 *
 * @param dataMap          hex-encoded caller-held DataMap
 * @param storageCostAtto  total storage cost paid in atto tokens ("0" if all chunks already existed)
 * @param gasCostWei       total gas cost paid in wei as a decimal string
 * @param chunksStored     number of chunks stored on the network
 * @param paymentModeUsed  which payment mode was actually used: "auto", "merkle", or "single"
 */
public record FilePutResult(
        String dataMap,
        String storageCostAtto,
        String gasCostWei,
        long chunksStored,
        String paymentModeUsed) {}
